apps/checkout-app/frontend/components/templates/ChannelDetails/ChannelDetails.tsx

```diff
@@ -45,71 +80,114 @@ const ChannelDetails: React.FC<ChannelDetailsProps> = ({
-                  </OffsettedListItem>
-                ))}
-              </OffsettedListBody>
-            </OffsettedList>
-          </AccordionDetails>
-        </Accordion>
-      ))}
-    </AppLayout>
+    <form>
+      <AppLayout
+        title={channelPaymentOptions.channel.name}
+        onBackClick={onBackClick}
+        onSettingsClick={onSettingsClick}
+        items={mapNodesToItems(channels)}
+        selectedItem={mapNodeToItem(channelPaymentOptions.channel)}
+        loading={loading}
+        onItemClick={onChannelClick}
+      >
+        <Typography variant="subtitle1">
+          <FormattedMessage {...messages.selectPaymentMethods} />
+        </Typography>
+        {loading ? (
+          <Skeleton />
+        ) : (
+          channelPaymentOptions.paymentOptions.map(
+            (paymentOption, paymentOptionIdx) => (
+              <Accordion
+                key={paymentOption.method.id}
+                className={classes.paymentOption}
+                elevation={0}
+              >
+                <AccordionSummary
+                  expandIcon={<ExpandMoreIcon />}
+                  className={classes.paymentOptionExpander}
+                >
+                  <div className={classes.paymentOptionIcon}></div>
+                  <Typography variant="subtitle2">
+                    {paymentOption.method.name}
+                  </Typography>
+                </AccordionSummary>
+                <AccordionDetails className={classes.paymentOptionDetails}>
+                  <OffsettedList gridTemplate={["1fr", actions(1)]}>
+                    <Controller
+                      key={paymentOption.id}
+                      name={paymentOption.id}
+                      control={control}
+                      defaultValue={getActivePaymentProvider(paymentOption)}
+                      render={({ field }) => (
+                        <OffsettedListBody>
+                          {paymentOption.availableProviders.map((provider) => (
+                            <OffsettedListItem
+                              key={provider.id}
+                              className={classes.paymentMethod}
+                            >
+                              <OffsettedListItemCell>
+                                {provider.label}
+                              </OffsettedListItemCell>
+                              <OffsettedListItemCell padding="action">
+                                <Switch
+                                  name={flattenSettingId(
+                                    paymentOptionIdx,
+                                    provider.id
+                                  )}
+                                  checked={field.value === provider.id}
+                                  onChange={() =>
+                                    field.onChange({
+                                      target: {
+                                        name: paymentOption.id,
```

Comment:
This correct? Should name and id be the same thing?

Reply:
Yes, ids differentiate the fields here